test/e2e/windows/gmsa.go

```diff
@@ -0,0 +1,158 @@
+					containers[i] = corev1.Container{
+						Name:  name,
+						Image: imageutils.GetPauseImageName(),
+					}
+				}
+
+				pod := &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: podName,
+						Annotations: map[string]string{
+							"pod.alpha.windows.kubernetes.io/gmsa-credential-spec":                         generateDummyCredSpecs(podDomain),
+							container2Name + ".container.alpha.windows.kubernetes.io/gmsa-credential-spec": generateDummyCredSpecs(container2Domain),
+						},
+					},
+					Spec: corev1.PodSpec{
+						Containers: containers,
+					},
+				}
+
+				ginkgo.By("creating a pod with correct GMSA annotations")
+				f.PodClient().Create(pod)
+
+				ginkgo.By("waiting for the pod and its containers to be running")
+				gomega.Eventually(func() bool {
+					pod, err := f.PodClient().Get(podName, metav1.GetOptions{})
+					if err != nil && pod.Status.Phase != corev1.PodRunning {
+						return false
+					}
+
+					for _, containerStatus := range pod.Status.ContainerStatuses {
+						if containerStatus.State.Running == nil {
+							return false
+						}
+					}
+
+					return true
+				}, 5*time.Minute, 1*time.Second).Should(gomega.BeTrue())
+
+				ginkgo.By("checking the domain reported by nltest in the containers")
+				namespaceOption := fmt.Sprintf("--namespace=%s", f.Namespace.Name)
+				for containerName, domain := range map[string]string{
+					container1Name: podDomain,
+					container2Name: container2Domain,
+				} {
+					var (
+						output string
+						err    error
+					)
+
+					containerOption := fmt.Sprintf("--container=%s", containerName)
+					// even for bogus creds, `nltest /PARENTDOMAIN` simply returns the AD domain, which is enough for our purpose here.
+					// note that the "eventually" part seems to be needed to account for the fact that powershell containers
+					// are a bit slow to become responsive, even when docker reports them as running.
+					gomega.Eventually(func() bool {
+						output, err = framework.RunKubectl("exec", namespaceOption, podName, containerOption, "--", "nltest", "/PARENTDOMAIN")
+						return err == nil
+					}, 1*time.Minute, 1*time.Second).Should(gomega.BeTrue())
+
+					if !strings.HasPrefix(output, domain) {
+						framework.Failf("Expected %q to start with %q", output, domain)
+					}
+
+					expectedSubstr := "The command completed successfully"
+					if !strings.Contains(output, expectedSubstr) {
+						framework.Failf("Expected %q to contain %q", output, expectedSubstr)
+					}
+				}
+
+				// If this was an e2e_node test, we could also check that the registry keys used to pass down the cred specs to Docker
+				// have been properly cleaned up - but as of right now, e2e_node tests don't support Windows. We should migrate this
+				// test to an e2e_node test when they start supporting Windows.
+
+				// FIXME: remove this before merging - this is here to check if this test is actually run
```

Comment:
This needs to be removed/fixed right?

Reply:
Yes - but before I remove it I'd like to see it failing where it's supposed to be running, to make sure this test actually runs somewhere... And I haven't seen that yet?

Reply:
Discussed with @wk8 offline that these tests won't run as PROW or get picked up by the other testgrid suites right now.